frontend/src/core/components/tools/compare/hooks/useCompareChangeNavigation.ts

```diff
@@ -0,0 +1,212 @@
+import { RefObject, useCallback } from 'react';
+
+type Pane = 'base' | 'comparison';
+
+export const useCompareChangeNavigation = (
+  baseScrollRef: RefObject<HTMLDivElement | null>,
+  comparisonScrollRef: RefObject<HTMLDivElement | null>
+) => {
+  return useCallback(
+    (changeValue: string, pane: Pane, pageNumber?: number) => {
+      const targetRef = pane === 'base' ? baseScrollRef : comparisonScrollRef;
+      const container = targetRef.current;
+      if (!container) {
+        return;
+      }
+
+      const findNodes = (): HTMLElement[] => {
+        return Array.from(
+          container.querySelectorAll(`[data-change-id="${changeValue}"]`)
+        ) as HTMLElement[];
+      };
+
+      const scrollToPageIfNeeded = () => {
+        if (!pageNumber) return false;
+        const pageEl = container.querySelector(
+          `.compare-diff-page[data-page-number="${pageNumber}"]`
+        ) as HTMLElement | null;
+        if (!pageEl) return false;
+        const top = Math.max(0, pageEl.offsetTop - 4);
+        container.dataset.programmatic = '1';
+        container.scrollTop = Math.max(0, top);
+        // also move peer immediately to the same page top to keep alignment
+        const peerRef = pane === 'base' ? comparisonScrollRef : baseScrollRef;
+        const peer = peerRef.current;
+        if (peer) {
+          const peerPageEl = peer.querySelector(
+            `.compare-diff-page[data-page-number="${pageNumber}"]`
+          ) as HTMLElement | null;
+          if (peerPageEl) {
+            const peerTop = Math.max(0, peerPageEl.offsetTop - 4);
+            peer.dataset.programmatic = '1';
+            peer.scrollTop = peerTop;
+          }
+        }
+        requestAnimationFrame(() => {
+          delete (container as any).dataset.programmatic;
+          if (peerRef.current) delete (peerRef.current as any).dataset.programmatic;
```

Comment:
```suggestion
          delete container.dataset.programmatic;
          if (peerRef.current) delete peerRef.current.dataset.programmatic;
```
Why do we not know what these are?